app/src/components/network/NetworkList.vue

```diff
@@ -43,19 +40,24 @@ export default {
   },
   computed: {
     ...mapState(["session"]),
-    ...mapGetters({ networkId: `network` }),
-    whichFlow() {	
-      if (this.$route.name === "select-network-recover") {	
-        return `/recover`	
-      } else if (this.$route.name === "select-network-create") {	
-        return `/create`	
-      } else {	
-        return ``	
-      }	
+    ...mapGetters([{ networkId: `network` }, `isExtension`]),
+    whichFlow() {
+      if (this.$route.name === "select-network-recover") {
+        return `/recover`
+      } else if (this.$route.name === "select-network-create") {
+        return `/create`
+      } else {
+        return ``
+      }
     },
   },
   methods: {
     async selectNetworkHandler(network) {
+      if (this.isExtension) {
+        if (this.networkId !== network.id) {
+          this.$store.dispatch(`setNetwork`, network)
```

Comment:
await?

Reply:
true, it should be actually await. I think in the separate PR it is alright, so there should be a conflict now

Reply:
Done